src/journal-remote/journal-gatewayd.c

```diff
@@ -307,42 +345,99 @@ static int request_parse_range(
         if (!colon)
                 m->cursor = strdup(range);
         else {
-                const char *p;
+                r = request_parse_range_skip_and_n_entries(m, colon);
+                if (r < 0)
+                        return r;
 
-                colon2 = strchr(colon + 1, ':');
-                if (colon2) {
-                        _cleanup_free_ char *t;
+                m->cursor = strndup(range, colon - range);
+        }
 
-                        t = strndup(colon + 1, colon2 - colon - 1);
-                        if (!t)
-                                return -ENOMEM;
+        if (!m->cursor)
+                return -ENOMEM;
 
-                        r = safe_atoi64(t, &m->n_skip);
-                        if (r < 0)
-                                return r;
-                }
+        m->cursor[strcspn(m->cursor, WHITESPACE)] = 0;
+        if (isempty(m->cursor))
+                m->cursor = mfree(m->cursor);
 
-                p = (colon2 ? colon2 : colon) + 1;
-                if (*p) {
-                        r = safe_atou64(p, &m->n_entries);
-                        if (r < 0)
-                                return r;
+        return 0;
+}
 
-                        if (m->n_entries <= 0)
-                                return -EINVAL;
 
-                        m->n_entries_set = true;
-                }
+static int request_parse_range_time(
+                RequestMeta *m,
+                const char *range) {
 
-                m->cursor = strndup(range, colon - range);
+        char *colon, *until;
+        int r;
+
+        range += 5;
+        range += strspn(range, WHITESPACE);
+
+        colon = strchr(range, ':');
+        if (!colon)
+                return -EINVAL;
+
+        if (colon - range > 0) {
+                _cleanup_free_ char *t;
+                t = strndup(range, colon - range);
+                if (!t)
+                        return -ENOMEM;
+
+                r = safe_atou64(t, &m->since);
+                if (r < 0)
+                        return r;
+
+                m->since_set = true;
         }
 
-        if (!m->cursor)
+        range = colon;
+        colon = strchr(range + 1, ':');
+        if (!colon)
+                until = strdup(range + 1);
+        else {
+                r = request_parse_range_skip_and_n_entries(m, colon);
+                if (r < 0)
+                        return r;
+
+                until = strndup(range + 1, colon - range - 1);
+        }
+
+        if (!until)
                 return -ENOMEM;
 
-        m->cursor[strcspn(m->cursor, WHITESPACE)] = 0;
-        if (isempty(m->cursor))
-                m->cursor = mfree(m->cursor);
+        until[strcspn(until, WHITESPACE)] = 0;
```

Comment:
Is this necessary?

Reply:
With this line : `range: time=1618136100000000:1618136400000000   :5` is accepted, while it is not without (because of the spaces between the second micro-ts and the colon). There's no interest in accepting values outside of the spec, so I removed it.